### DISCUSSION TITLE
cFE App Developers Guide: does the Interrupt Management Library exist?

### DISCUSSION THREAD

**Discussion by jeff121271** [nasa/cFS#296] [Q&A] [answered]
Hi all,

I need to write a hardware-servicing app, and the setup described in [section 4.1.2](https://github.com/nasa/cFE/blob/main/docs/cFE%20Application%20Developers%20Guide.md#412-hardware-servicing-application) of the cFE App Developers' Guide looks like exactly what I need.
I'm having some trouble establishing where exactly the interrupt management library lives, and indeed if it actually exists. Is this library available anywhere, or is it just described as future functionality here?

**Comment by astrogeco** [comment, marked as answer]
Hi, thanks for the question and welcome to the community! 

Unfortunately the interrupt APIs were deprecated and it looks like we missed the update to that diagram. That being said, there might be other ways to do what you need. 

See these issues for more context

https://github.com/nasa/osal/issues/385
https://github.com/nasa/cFE/issues/571
https://github.com/nasa/PSP/issues/180

**Reply by jeff121271** [reply]
Okay, that makes sense. Thanks for the quick answer!

**Reply by skliper** [reply]
Opened an issue to update the cFE App Developers Guide: https://github.com/nasa/cFE/issues/1658
